### PR TITLE
Platformize configuration-cache subproject

### PR DESCRIPTION
**Pull request by alllex**
Moved `subprojects/configuration-cache` to `platforms/core-configuration/configuration-cache`.

### CODE REVIEW

**Comment by alllex** [comment]
@bot-gradle test this

File: .cm/platform_labels.cm [hunk 1]
```diff
@@ -110,9 +110,9 @@ platforms:
   - cc:
     name: 'configuration_cache'
     subprojects:
+      - 'platforms/core-configuration/configuration-cache/'
```
Comment (tresat) [inline .cm/platform_labels.cm:113]: ❗ If the platform is going to be named `core-configuration`, then this should look like:

```
  - cc:
    name: 'core_configuration'
    subprojects:
      - 'platforms/core-configuration'
      - 'subprojects/api-metadata/'
...continues
```

There's no need to be more specific, everything under the `platforms/core-configuration` directory will match.

The `summary_table.cm` file should be adjusted to match this as well.

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -18,9 +18,9 @@ subprojects/build-operations/               @gradle/bt-execution @gradle/bt-ge-b
 subprojects/build-profile/                  @gradle/bt-execution
 
 # Core automation platform (core/configuration)
+platforms/core-configuration/configuration-cache/       @gradle/bt-configuration
```
Comment (tresat) [inline .github/CODEOWNERS:21]:
```suggestion
platforms/core-configuration/       @gradle/bt-configuration
```

**Review by tresat — CHANGES_REQUESTED** [review]

File: .github/CODEOWNERS (hunk 2 above)
Comment (alllex) [reply .github/CODEOWNERS]: Done

File: .cm/platform_labels.cm (hunk 1 above)
Comment (alllex) [reply .cm/platform_labels.cm]: Updated

**Comment by alllex** [comment]
@bot-gradle test this

**Review by tresat — APPROVED** [review]

File: .cm/platform_labels.cm [hunk 3]
```diff
@@ -107,10 +109,10 @@ platforms:
       - 'subprojects/internal-integ-testing/'
       - 'subprojects/internal-performance-testing/'
       - 'subprojects/internal-testing/'
-  - cc:
-    name: 'configuration_cache'
+  - configuration:
+    name: 'configuration'
```
Comment (mlopatkin) [inline .cm/platform_labels.cm:113]: :question: this seems to be inconsistent with other `core/` labels, like `core_execution` and `core_runtime`.

File: .cm/summary_table.cm [hunk 4]
```diff
@@ -122,12 +124,12 @@ platforms:
       - 'subprojects/internal-integ-testing/'
       - 'subprojects/internal-performance-testing/'
       - 'subprojects/internal-testing/'
-  - cc:
-    name: 'configuration_cache'
+  - configuration:
+    name: 'configuration'
```
Comment (mlopatkin) [inline .cm/summary_table.cm:128]: ❓ this seems to be inconsistent with other `core/` labels, like `core_execution` and `core_runtime`.

**Review by mlopatkin — COMMENTED** [review]
Thank you for moving this forward!

Only one concern: is the inconsistent label naming intended?

File: .cm/platform_labels.cm (hunk 3 above)
Comment (alllex) [reply .cm/platform_labels.cm]: Fixed

File: .cm/summary_table.cm (hunk 4 above)
Comment (alllex) [reply .cm/summary_table.cm]: Fixed. See the other comment.

**Review by mlopatkin — APPROVED** [review]

**Comment by alllex** [comment]
@bot-gradle test and merge

**Comment by alllex** [comment]
@bot-gradle test and merge

**Pull request merged**